iothub/device/src/Exceptions/IotHubStatusCode.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.Devices.Client.Exceptions
+{
+    /// <summary>
+    /// The IoT Hub status code.
+    /// </summary>
+    public enum IotHubStatusCode
```

Comment:
To replace the [particular ](https://github.com/Azure/azure-iot-sdk-csharp/tree/previews/v2/iothub/device/src/Exceptions)exception types, we use the IotHubStatusCode enum here and expose to users via the class `IotHubClientException`